torch/nn/modules/module.py

```diff
@@ -249,26 +249,41 @@ def forward(self, x):
     the change."""
 
     training: bool
+    _parameters: Dict[str, Optional[Parameter]]
+    _buffers: Dict[str, Optional[Tensor]]
+    _non_persistent_buffers_set: Set[str]
+    _backward_hooks: Dict[int, Callable]
     _is_full_backward_hook: Optional[bool]
+    _forward_hooks: Dict[int, Callable]
+    _forward_pre_hooks: Dict[int, Callable]
+    _state_dict_hooks: Dict[int, Callable]
+    _load_state_dict_pre_hooks: Dict[int, Callable]
+    _load_state_dict_post_hooks: Dict[int, Callable]
+    _modules: Dict[str, Optional['Module']]
 
     def __init__(self) -> None:
         """
         Initializes internal Module state, shared by both nn.Module and ScriptModule.
+
+        Calls object.__setattr__(self, 'a', a) instead of the typical self.a = a
+        to avoid Module.__setattr__ overhead. Module's __setattr__ has special
+        handling for parameters, submodules, and buffers but simply calls into
+        object.__setattr__ for all other attributes.
```

Comment:
I suggest keeping this information in a private comment below rather than adding it to the init docstring. The implementation details are a bit out-of-scope for public docs.